modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/DistributionZoneSqlDdlParserTest.java

```diff
@@ -48,48 +48,89 @@ public class DistributionZoneSqlDdlParserTest extends AbstractParserTest {
+        expectUnparsed(createZone, "CREATE ZONE \"public\".\"test_Zone\" \"STORAGE PROFILES\"['p']");
 
-        createZone = parseCreateZone("create zone \"public-test_Zone\"");
+        createZone = parseCreateZoneV2("create zone \"public-test_Zone\" storage profiles['p']");
         assertThat(createZone.name().names, is(List.of("public-test_Zone")));
-        expectUnparsed(createZone, "CREATE ZONE \"public-test_Zone\"");
+        expectUnparsed(createZone, "CREATE ZONE \"public-test_Zone\" \"STORAGE PROFILES\"['p']");
     }
 
     /**
      * Parse CREATE ZONE IF NOT EXISTS statement.
      */
     @Test
     public void createZoneIfNotExists() {
-        IgniteSqlCreateZone createZone = parseCreateZone("create zone if not exists test_zone");
+        IgniteSqlCreateZoneV2 createZone = parseCreateZoneV2("create zone if not exists test_zone storage profiles['p']");
 
         assertTrue(createZone.ifNotExists());
         assertNull(createZone.createOptionList());
 
-        expectUnparsed(createZone, "CREATE ZONE IF NOT EXISTS \"TEST_ZONE\"");
+        expectUnparsed(createZone, "CREATE ZONE IF NOT EXISTS \"TEST_ZONE\" \"STORAGE PROFILES\"['p']");
+    }
+
+    @Test
+    public void createZoneWithOptions() {
+        IgniteSqlCreateZoneV2 createZone = parseCreateZoneV2(
+                "create zone test_zone "
+                        + "(replicas 2, "
+                        + "partitions 3, "
+                        + "nodes filter '(\"US\" || \"EU\") && \"SSD\"', "
+                        + "distribution algorithm 'test_Distribution', "
+                        + "auto adjust 1, "
+                        + "auto scale up 2, "
+                        + "auto scale down 3, "
+                        + "consistency mode 'HIGH_AVAILABILITY') "
+                        + "storage profiles ['default', 'new']"
+        );
+
+        assertNotNull(createZone.createOptionList());
+
+        List<SqlNode> optList = createZone.createOptionList().getList();
+
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.REPLICAS, 2);
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.PARTITIONS, 3);
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.DISTRIBUTION_ALGORITHM, "test_Distribution");
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.DATA_NODES_FILTER, "(\"US\" || \"EU\") && \"SSD\"");
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.DATA_NODES_AUTO_ADJUST, 1);
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.DATA_NODES_AUTO_ADJUST_SCALE_UP, 2);
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.DATA_NODES_AUTO_ADJUST_SCALE_DOWN, 3);
+        assertThatZoneOptionPresent(optList, ZoneOptionEnum.CONSISTENCY_MODE, "HIGH_AVAILABILITY");
+
+        expectUnparsed(createZone, "CREATE ZONE \"TEST_ZONE\" ("
+                + "\"REPLICAS\" 2, "
+                + "\"PARTITIONS\" 3, "
+                + "\"NODES FILTER\" '(\"US\" || \"EU\") && \"SSD\"', "
```

Comment:
`unparse` must return string representation of AST, i.e. you should be able to parse result of unparse, and the resulting AST must be equal to original one